### PR TITLE
disable some unsupported syntax

### PR DESCRIPTION
**Pull request by yangzhg**
#3373
disable some syntax when subquery is not in binary predicate in case when clause

### CODE REVIEW

**Comment by morningman** [comment]
Hi @yangzhg , you need to create an issue to describe your problem, better to give an example about the query you try to disable.

**Comment by yangzhg** [comment]
> Hi @yangzhg , you need to create an issue to describe your problem, better to give an example about the query you try to disable.

done

File: fe/src/main/java/org/apache/doris/analysis/SelectStmt.java [hunk 1]
```diff
@@ -1362,9 +1362,24 @@ private void rewriteSelectList(ExprRewriter rewriter) throws AnalysisException {
...
+            CaseExpr caseExpr = (CaseExpr) item.getExpr();
+
+            int childIdx = 0;
+            if (caseExpr.hasCaseExpr()
+                    && caseExpr.getChild(childIdx++).contains(Predicates.instanceOf(Subquery.class))) {
+                throw new AnalysisException("Only support subquery in binary predicate in case statement.");
+            }
+            while (childIdx + 2 <= caseExpr.getChildren().size()) {
+                Expr child = caseExpr.getChild(childIdx++);
+                // case
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/analysis/SelectStmt.java:1374]: `caseExpr.getChild(childIdx++)` here you got the `when` expr

File: fe/src/main/java/org/apache/doris/analysis/SelectStmt.java [hunk 2]
```diff
@@ -1362,9 +1362,24 @@ private void rewriteSelectList(ExprRewriter rewriter) throws AnalysisException {
...
+                if (!(child instanceof BinaryPredicate) && child.contains(Predicates.instanceOf(Subquery.class))) {
+                    throw new AnalysisException("Only support subquery in binary predicate in case statement.");
+                }
+                // when
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/analysis/SelectStmt.java:1378]: Here is to skip `then` expr, not `when`

**Comment by kangkaisen** [comment]
@yangzhg Please add a UT

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**